versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/AbstractMergeTransplant.java

```diff
@@ -64,24 +78,119 @@ void transplant() throws Exception {
         mergeTransplant(
             (target, expectedHead, branch, commitHashes, i) ->
                 databaseAdapter.transplant(
-                    target, expectedHead, Arrays.asList(commitHashes).subList(0, i + 1)));
+                    target,
+                    expectedHead,
+                    Arrays.asList(commitHashes).subList(0, i + 1),
+                    Function.identity()));
 
     BranchName conflict = BranchName.of("conflict");
 
     // no conflict, when transplanting the commits from against the current HEAD of the
     // conflict-branch
     Hash noConflictHead = databaseAdapter.toHash(conflict);
-    databaseAdapter.transplant(conflict, Optional.of(noConflictHead), Arrays.asList(commits));
+    databaseAdapter.transplant(
+        conflict, Optional.of(noConflictHead), Arrays.asList(commits), Function.identity());
 
     // again, no conflict (same as above, just again)
-    databaseAdapter.transplant(conflict, Optional.empty(), Arrays.asList(commits));
+    databaseAdapter.transplant(
+        conflict, Optional.empty(), Arrays.asList(commits), Function.identity());
 
     assertThatThrownBy(
-            () -> databaseAdapter.transplant(conflict, Optional.empty(), Collections.emptyList()))
+            () ->
+                databaseAdapter.transplant(
+                    conflict, Optional.empty(), Collections.emptyList(), Function.identity()))
         .isInstanceOf(IllegalArgumentException.class)
         .hasMessage("No hashes to transplant given.");
   }
 
+  @Test
```

Comment:
Both new tests duplicate a lot of code from `merg()` and `transplant()` and should be refactored to parameterized tests.